### PR TITLE
feat(toolkits/pro) An error message is displayed when the dependency installation fails

### LINKED ISSUE #116: 🐛 [Bug]: 安装eggJs服务端依赖的时候，因为网络环境的原因，安装报错，但提示显示安装成功了，
### Version

latest

### node-version

latest

### Link to minimal reproduction

_No response_

### Step to reproduce

image

### What is expected

安装报错，显示安装失败的提示

### What is actually happening

_No response_

### Any additional comments (optional)

_No response_

### PR DESCRIPTION
**Pull request by liberty-zlk**
…installation fails

# PR
fix 1 issues:
fix #116 

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-cli/blob/main/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/toolkits/pro/src/lib/init.ts [hunk 1]
```diff
@@ -384,7 +397,6 @@ export default async () => {
   try {
     installDependencies(projectInfo);
   } catch (e) {
-    log.error('npm 依赖安装失败');
```
Comment (fengyon) [inline packages/toolkits/pro/src/lib/init.ts]: 这里的代码不需要删除，执行命令可能会因为环境等问题失败，还是需要提示的

File: packages/toolkits/pro/src/lib/init.ts [hunk 2]
```diff
@@ -291,19 +291,32 @@ export const installDependencies = (answers: ProjectInfo) => {
...
-  spawn.sync('npm', ['install'], {
+  const installClientResult = spawn.sync('npm', ['install'], {
     cwd: serverConfirm ? `${name}/web` : `${name}/`,
     stdio: 'inherit',
   });
-  log.success('客户端 npm 依赖安装成功');
+  if(installClientResult.status === 0) {
+    log.success('客户端 npm 依赖安装成功');
+  }else {
+    log.error('客户端 npm 依赖安装失败');
```
Comment (fengyon) [inline packages/toolkits/pro/src/lib/init.ts]: 这里把 `installClientResult`的错误 抛出一下即可，由调用者处理错误并提示
```
throw installClientResult.xxx
```

File: packages/toolkits/pro/src/lib/init.ts [hunk 3]
```diff
@@ -291,19 +291,32 @@ export const installDependencies = (answers: ProjectInfo) => {
...
-    spawn.sync('npm', ['install'], {
+    const installServiceResult = spawn.sync('npm', ['install'], {
       cwd: `${name}/${serverFramework}/`,
       stdio: 'inherit',
     });
-    log.success('服务端 npm 依赖安装成功');
+    if(installServiceResult.status === 0) {
+      log.success('服务端 npm 依赖安装成功');
+    }else {
+      log.error('服务端 npm 依赖安装失败');
```
Comment (fengyon) [inline packages/toolkits/pro/src/lib/init.ts]: 这里把 `installClientResult`的错误 抛出一下即可，由调用者处理错误并提示
```
throw installClientResult.xxx
```

File: packages/toolkits/pro/src/lib/init.ts (hunk 1 above)
Comment (liberty-zlk) [reply packages/toolkits/pro/src/lib/init.ts]: > 这里的代码不需要删除，执行命令可能会因为环境等问题失败，还是需要提示的

已改

File: packages/toolkits/pro/src/lib/init.ts (hunk 2 above)
Comment (liberty-zlk) [reply packages/toolkits/pro/src/lib/init.ts]: > 这里把 `installClientResult`的错误 抛出一下即可，由调用者处理错误并提示
> 
> ```
> throw installClientResult.xxx
> ```

已改

File: packages/toolkits/pro/src/lib/init.ts (hunk 3 above)
Comment (liberty-zlk) [reply packages/toolkits/pro/src/lib/init.ts]: > 这里把 `installClientResult`的错误 抛出一下即可，由调用者处理错误并提示
> 
> ```
> throw installClientResult.xxx
> ```

已改

**Review by fengyon — APPROVED** [review]
检视通过

**Pull request merged**